crates/turborepo-lib/src/globwatcher/mod.rs

```diff
@@ -0,0 +1,166 @@
+use std::{
+    collections::{HashMap, HashSet},
+    path::PathBuf,
+    sync::{Arc, Mutex},
+};
+
+use futures::StreamExt;
+use globwatch::{GlobSender, GlobWatcher, StopToken, Watcher};
+use itertools::Itertools;
+use log::{trace, warn};
+use notify::RecommendedWatcher;
+
+/// Tracks changes for a given hash. A hash is a unique identifier for a set of
+/// files. Given a hash and a set of globs to track, this will watch for file
+/// changes and allow the user to query for changes.
+#[derive(Clone)]
+pub struct HashGlobWatcher<T: Watcher> {
+    hash_globs: Arc<Mutex<HashMap<String, Glob>>>,
+    glob_status: Arc<Mutex<HashMap<String, HashSet<String>>>>,
```

Comment:
Should the keys to the maps be `Rc<String>` since they're duplicated between `hash_globs` and `glob_status`?

Reply:
I like it.